src/python/pants/engine/http.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Dict, Optional, Tuple
+
+import requests
+
+from pants.engine.rules import side_effecting
+
+
+@dataclass(frozen=True)
+class HttpGetResponse:
+  url: str
+  status_code: Optional[int]
+  output_bytes: Optional[bytes]
+  headers: Tuple[Tuple[str, str], ...]
+
+
+@side_effecting
+@dataclass(frozen=True)
+class HttpRequester:
+
+  def get_request(self, url: str, *, headers: Dict[str, str] = {}) -> HttpGetResponse:
```

Comment:
@Eric-Arellano we should enable [flake8-mutable](https://github.com/ebeweber/flake8-mutable) on this repo...

Reply:
Agreed. We need to finish the `flake8-pantsbuild` plugin to be able to start using Flake8 in the Pants repo, which I'm hoping to finish by end of the month 🤞 (it's a side project).